src/TraceEvent/Computers/TraceManagedProcess.cs

```diff
@@ -1010,6 +1010,59 @@ internal static void SetupCallbacks(TraceEventDispatcher source)
                             finalizationCount + 1 :
                             1;
                 };
+
+                source.Clr.TypeBulkType += data =>
+                {
+                    var stats = currentManagedProcess(data);
+                    var typeMapping = stats.GC.m_stats.TypeMapping;
+
+                    var clonedData = data;
+                    clonedData = (GCBulkTypeTraceData)clonedData.Clone();
+
+                    var count = clonedData.Count;
+                    for (int i = 0; i < count; i++)
+                    {
+                        var bulkTypeValues = clonedData.Values(i);
+                        var typeId = bulkTypeValues.TypeID;
+                        if (typeMapping.ContainsKey(typeId))
+                            continue;
+
+                        typeMapping[typeId] = bulkTypeValues;
+                    }
+                };
+
+                source.Clr.GCFinalizeObject += data =>
```

Comment:
Roughly speaking there is no relationship between this code and the code you referred to.   The code you referenced is in the 'first pass' when ETL is converted t ETLX, and the 'doNothing' delegate is there only to insure that during this first pass we parse the event during that pass.   The code you have here is an independent pass over the events.